src/Http/Cookie/SameSiteEnum.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Cookie;
+
+use ValueError;
+
+// phpcs:ignoreFile
+enum SameSiteEnum: string
+{
+    case Lax = 'Lax';
+    case Strict = 'Strict';
+    case None = 'None';
+
+    /**
+     * Get enum instance for the given name.
+     *
+     * @param string $name The name to map to an enum case.
+     * @return static
+     * @throws \ValueError If there is no matching case defined.
+     */
+    public static function fromName(string $name): static
```

Comment:
I have added these `fromName()` and `fromTryName()` methods based on the ones seen here https://github.com/alexanderpas/php-http-enum/blob/master/src/Method.php.

PHP doesn't allow overriding the `BackedEnum::from()` and `BackedEnum::tryFrom()` methods.

Reply:
Will we need these on every enum we have? That seems like a lot of code to support a not-significantly better API than what we get with constants, and an array. We could even make the array value private so it can't be changed.

Doing this much for every potential enum in the framework feels like a lot of code.

Reply:
This is mainly to retain the ability to use case-insensitive strings for samesite values in `Cookie`. But I guess we will eventually drop support for strings so I will keep the string casing manipulation in `Cookie` itself and keep the enum class clean.